packages/frontend/src/components/RegisterForms/VerifyEmailForm.tsx

```diff
@@ -0,0 +1,129 @@
+            code: isNotEmpty('This field is required.'),
+        },
+    });
+    const errorMessage = form.errors.code;
+    const expirationTime = data?.otp?.expiresAt || new Date();
+    const loadingState =
+        statusLoading || emailLoading || health.isLoading || isLoading;
+
+    useEffect(() => {
+        if (data?.otp && data?.otp.numberOfAttempts > 0) {
+            const remainingAttempts = 5 - data.otp.numberOfAttempts;
+            const message = data.otp.isExpired
+                ? 'Your one-time password expired. Please resend a verification email.'
+                : data.otp.numberOfAttempts < 5
+                ? `The code doesn't match the one we sent you. You have ${remainingAttempts} attempt${
+                      remainingAttempts > 1 ? 's' : ''
+                  } left.`
+                : "Hmm that code doesn't match the one we sent you. You've already had 5 attempts, please resend a verification email and try again.";
+            form.setFieldError('code', message);
+        } else {
+            form.clearFieldError('code');
+        }
+    }, [data, form.setFieldError, form.clearFieldError]);
+
+    if (loadingState) {
+        return <LoadingState title="" />;
+    }
+
+    return (
+        // FIXME: use Mantine sizes for width
+        <Stack spacing="md" justify="center" align="center" w={300} mx="auto">
+            <Title order={3}>Check your inbox!</Title>
+            <Text color="gray.6" ta="center">
+                Verify your email address by entering the code we've just sent
+                to <b>{user?.data?.email || 'your email'}</b>
+            </Text>
+            <form
+                name="verifyEmail"
+                onSubmit={form.onSubmit((values: { code: string }) =>
+                    verifyCode(values.code),
+                )}
+            >
+                <Stack spacing="xs" justify="center" align="center" mt="md">
+                    <PinInput
+                        aria-label="One-time password"
+                        name="code"
+                        length={6}
+                        oneTimeCode
+                        disabled={data?.otp?.isMaxAttempts}
+                        {...form.getInputProps('code')}
+                        data-testid="pin-input"
+                    />
+                    <Text ta="center" color="red.7">
+                        {errorMessage?.toString()}
+                    </Text>
+                </Stack>
+                <Countdown
+                    key={expirationTime?.toString()}
+                    date={expirationTime}
+                    renderer={({ minutes, seconds, completed }) => {
+                        if (completed || data?.otp?.isMaxAttempts) {
+                            return <></>;
+                        }
+                        return (
+                            // FIXME: use Mantine sizes for width
+                            <Stack spacing="xs" mt="md" w={290} align="center">
```

Comment:
you can remove w=290 if you add a proper padding to the first Stack

Reply:
this is needed because of width fluctuation with the countdown